frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts

```diff
@@ -133,13 +133,16 @@ const sortTasksByRunAfterAndFrom = (
     let flag = -1;
     if (conditions.hasRunAfterDependency(output[i])) {
       for (let j = 0; j < output.length; j++) {
-        if (i < j && output[j].taskRef.name === output[i].runAfter[output[i].runAfter.length - 1]) {
+        if (
+          i < j &&
+          output[j].taskRef?.name === output[i].runAfter[output[i].runAfter.length - 1]
```

Comment:
This sort function seems to support only taskRef, to handle taskSpec, we could  add  this fallback ` || t.name  === output[i].runAfter[output[i].runAfter.length - 1]` ? 
but I am wondering, regardless of this sorting function's output the final structure of the pipeline is already correct :thinking:

Reply:
yes right, I tried with different yamls but nowhere did the visualization break
cc @andrewballantyne ^